lib/interpreters/split_clustering.rb

```diff
@@ -129,6 +120,302 @@ def self.generate_split_vrps(service_vrp, _job = nil, block = nil)
       end
     end
 
+    def self.split_solve_candidate?(service_vrp)
+      vrp = service_vrp[:vrp]
+      if service_vrp[:split_level].nil?
+        empties_or_fills = vrp.services.select{ |s| s.quantities.any?(&:fill) || s.quantities.any?(&:empty) }
+        depot_ids = vrp.vehicles.flat_map{ |vehicle| [vehicle.start_point_id, vehicle.end_point_id].compact }.uniq
+
+        !vrp.scheduling? &&
+          vrp.preprocessing_max_split_size &&
+          vrp.vehicles.size > 1 &&
+          (vrp.resolution_vehicle_limit.nil? || vrp.resolution_vehicle_limit > 1) &&
+          (vrp.shipments.size + vrp.services.size - empties_or_fills.size) > vrp.preprocessing_max_split_size &&
```

Comment:
in fact each shipment acts as a "single" point inside clustering (since we cannot separate the pickup and delivery of a shipment) so I find this condition more clear which handles the extreme case of "max_split_size=1" naturally as well.

Reply:
I resolve this convo, feel free to unresolve it if you want to discuss further